kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/spec/interceptor/instance/AfterSpecListenerSpecInterceptor.kt

```diff
@@ -0,0 +1,36 @@
+package io.kotest.engine.spec.interceptor.instance
+
+import io.kotest.core.config.ExtensionRegistry
+import io.kotest.core.listeners.AfterSpecListener
+import io.kotest.core.spec.Spec
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import io.kotest.engine.flatMap
+import io.kotest.engine.spec.SpecExtensions
+import io.kotest.engine.spec.interceptor.SpecInterceptor
+
+/**
+ * Invokes any [AfterSpecListener] callbacks for the given spec.
+ * These listeners are only invoked if the spec has at least one defined test AND all defined tests are not ignored.
+ */
+internal class AfterSpecListenerSpecInterceptor(private val registry: ExtensionRegistry) : SpecInterceptor {
+   override suspend fun intercept(
+      spec: Spec,
+      fn: suspend (Spec) -> Result<Map<TestCase, TestResult>>
+   ): Result<Map<TestCase, TestResult>> {
+      return fn(spec).flatMap { results ->
+         if (hasActiveTest(results)) {
+            SpecExtensions(registry)
+               .afterSpec(spec)
+               .map { results }
+         } else {
+            Result.success(results)
+         }
+      }
+   }
+
+   // the spec results are considered active if at least test has a non ignored status
+   private fun hasActiveTest(results: Map<TestCase, TestResult>): Boolean {
+      return results.any { it.value !is TestResult.Ignored }
```

Comment:
I'm wondering if we should actually do this.

If one decides to ignore all tests in a class just to see what the database is doing or if one prepares the database before writing tests (empty list), this would be a non-intuitive issue

Reply:
I know this is a fairly far-fetched what-if, but I don't see it as thaaat unlikely and can think of similar examples

Reply:
If we are to keep this, we should probably do the same for BeforeSpec listeners

Reply:
BeforeSpec already does this, so this is aligning AfterSpec with the same behavior.